custom_components/tahoma/climate_devices/atlantic_pass_apc_heating_zone.py

```diff
@@ -0,0 +1,334 @@
+COMMAND_REFRESH_TARGET_TEMPERATURE = "refreshTargetTemperature"
+COMMAND_SET_COMFORT_HEATING_TARGET_TEMPERATURE = "setComfortHeatingTargetTemperature"
+COMMAND_SET_DEROGATION_ON_OFF_STATE = "setDerogationOnOffState"
+COMMAND_SET_DEROGATED_TARGET_TEMPERATURE = "setDerogatedTargetTemperature"
+COMMAND_SET_DEROGATION_TIME = "setDerogationTime"
+COMMAND_SET_ECO_HEATING_TARGET_TEMPERATURE = "setEcoHeatingTargetTemperature"
+COMMAND_SET_OPERATING_MODE = "setOperatingMode"
+COMMAND_SET_PASS_APC_HEATING_MODE = "setPassAPCHeatingMode"
+COMMAND_SET_TARGET_TEMPERATURE = "setTargetTemperature"
+
+CORE_COMFORT_HEATING_TARGET_TEMPERATURE_STATE = (
+    "core:ComfortHeatingTargetTemperatureState"
+)
+CORE_DEROGATED_TARGET_TEMPERATURE_STATE = "core:DerogatedTargetTemperatureState"
+CORE_DEROGATION_ON_OFF_STATE = "core:DerogationOnOffState"
+CORE_ECO_HEATING_TARGET_TEMPERATURE_STATE = "core:EcoHeatingTargetTemperatureState"
+CORE_HEATING_ON_OFF_STATE = "core:HeatingOnOffState"
+CORE_TARGET_TEMPERATURE_STATE = "core:TargetTemperatureState"
+
+IO_DEROGATION_REMAINING_TIME_STATE = "io:DerogationRemainingTimeState"
+IO_PASS_APC_HEATING_MODE_STATE = "io:PassAPCHeatingModeState"
+IO_PASS_APC_HEATING_PROFILE_STATE = "io:PassAPCHeatingProfileState"
+IO_TARGET_HEATING_LEVEL_STATE = "io:TargetHeatingLevelState"
+
+PASS_APC_HEATING_MODE_STATE_ABSENCE = "absence"
+PASS_APC_HEATING_MODE_STATE_COMFORT = "comfort"
+PASS_APC_HEATING_MODE_STATE_DEROGATION = "derogation"
+PASS_APC_HEATING_MODE_STATE_INTERNAL_SCHEDULING = "internalScheduling"
+PASS_APC_HEATING_MODE_STATE_STOP = "stop"
+PASS_APC_HEATING_PROFILE_STATE_ABSENCE = "absence"
+PASS_APC_HEATING_PROFILE_STATE_COMFORT = "comfort"
+PASS_APC_HEATING_PROFILE_STATE_DEROGATION = "derogation"
+PASS_APC_HEATING_PROFILE_STATE_ECO = "eco"
+PASS_APC_HEATING_PROFILE_STATE_INTERNAL_SCHEDULING = "internalScheduling"
+PASS_APC_HEATING_PROFILE_STATE_STOP = "stop"
+
+MAP_PRESET_MODES = {
+    PASS_APC_HEATING_PROFILE_STATE_ECO: PRESET_ECO,
+    PASS_APC_HEATING_PROFILE_STATE_COMFORT: PRESET_COMFORT,
+    PASS_APC_HEATING_PROFILE_STATE_INTERNAL_SCHEDULING: CUSTOM_PRESET_AUTO,
+    PASS_APC_HEATING_PROFILE_STATE_DEROGATION: CUSTOM_PRESET_DEROGATION,
+    PASS_APC_HEATING_PROFILE_STATE_STOP: CUSTOM_PRESET_STOP,
+    PASS_APC_HEATING_PROFILE_STATE_ABSENCE: PRESET_AWAY,
+}
+MAP_REVERSE_PRESET_MODES = {v: k for k, v in MAP_PRESET_MODES.items()}
+
+
+class AtlanticPassAPCHeatingZone(OverkizEntity, ClimateEntity):
+    """Representation of Atlantic Pass APC Heating and Cooling Zone."""
+
+    _attr_hvac_modes = [HVAC_MODE_OFF, HVAC_MODE_HEAT, HVAC_MODE_AUTO]
+    _attr_supported_features = SUPPORT_PRESET_MODE | SUPPORT_TARGET_TEMPERATURE
+    _attr_temperature_unit = TEMP_CELSIUS
+
+    def __init__(self, device_url: str, coordinator: OverkizDataUpdateCoordinator):
+        """Init method."""
+        super().__init__(device_url, coordinator)
+
+        self._temp_sensor_entity_id = None
```

Comment:
Instead of this you can use the linked_device property. There is few examples. It will remove a lot of lines and ease the code.